server/src/test/java/org/elasticsearch/reservedstate/service/FileSettingsServiceTests.java

```diff
@@ -229,36 +220,77 @@ public void testStopWorksInMiddleOfProcessing() throws Exception {
+        assertTrue(fileSettingsService.watching());
 
-        Files.createDirectories(service.watchedFileDir());
+        Files.createDirectories(fileSettingsService.watchedFileDir());
 
         // Make some fake settings file to cause the file settings service to process it
-        writeTestFile(service.watchedFile(), "{}");
+        writeTestFile(fileSettingsService.watchedFile(), "{}");
 
         // we need to wait a bit, on MacOS it may take up to 10 seconds for the Java watcher service to notice the file,
         // on Linux is instantaneous. Windows is instantaneous too.
         assertTrue(processFileLatch.await(30, TimeUnit.SECONDS));
 
         // Stopping the service should interrupt the watcher thread, we should be able to stop
-        service.stop();
-        assertFalse(service.watching());
-        service.close();
+        fileSettingsService.stop();
+        assertFalse(fileSettingsService.watching());
+        fileSettingsService.close();
         // let the deadlocked thread end, so we can cleanly exit the test
         deadThreadLatch.countDown();
     }
 
+    public void testHandleSnapshotRestoreClearsMetadata() throws Exception {
+        ClusterState state = ClusterState.builder(clusterService.state())
+            .metadata(
+                Metadata.builder(clusterService.state().metadata())
+                    .put(new ReservedStateMetadata(FileSettingsService.NAMESPACE, 1L, Map.of(), null))
+                    .build()
+            )
+            .build();
+
+        Metadata.Builder metadata = Metadata.builder(state.metadata());
+        fileSettingsService.handleSnapshotRestore(state, metadata);
+
+        assertThat(metadata.build().reservedStateMetadata(), anEmptyMap());
+    }
+
+    public void testHandleSnapshotRestoreResetsMetadata() throws Exception {
+        fileSettingsService.start();
+        fileSettingsService.clusterChanged(new ClusterChangedEvent("test", clusterService.state(), ClusterState.EMPTY_STATE));
+
+        Files.createDirectories(fileSettingsService.watchedFileDir());
+        // contents of the JSON don't matter, we just need a file to exist
+        writeTestFile(fileSettingsService.watchedFile(), "{}");
+        assertTrue(fileSettingsService.watching());
+
+        ClusterState state = ClusterState.builder(clusterService.state())
+            .metadata(
+                Metadata.builder(clusterService.state().metadata())
+                    .put(new ReservedStateMetadata(FileSettingsService.NAMESPACE, 1L, Map.of(), null))
+                    .build()
+            )
+            .build();
+
+        Metadata.Builder metadata = Metadata.builder();
+        fileSettingsService.handleSnapshotRestore(state, metadata);
+
+        assertThat(
+            metadata.build().reservedStateMetadata(),
+            hasEntry(FileSettingsService.NAMESPACE, new ReservedStateMetadata(FileSettingsService.NAMESPACE, 0L, Map.of(), null))
```

Comment:
nit, for readability?
```suggestion
            hasEntry(FileSettingsService.NAMESPACE, new ReservedStateMetadata(FileSettingsService.NAMESPACE, ReservedStateMetadata.RESTORED_VERSION, Map.of(), null))
```